core/src/view/view.cpp

```diff
@@ -188,6 +188,8 @@ void View::screenToGroundPlane(float& _screenX, float& _screenY) const {
     
     _screenX = ray_world.x;
     _screenY = ray_world.y;
+    
+    return glm::vec2(t, ray_world.z);
```

Comment:
Conceptually we don't (negative `t` for ray to ground intersections are always behind), but updating the tile set should still occur even if one of the 4 viewport casted rays have intersections behind the camera, the only way to know is by looking up is by looking at the sign of `z` ray component.